src/components/modals/EditProfileModal.tsx

```diff
@@ -124,21 +134,32 @@ const EditProfileModal: React.FC<EditProfileModalProps> = ({
                             sx={{ width: 250, mb: 2 }}
                             onClick={changeUsername}
                         />
-                        <FormControl sx={{ m: 0.5 }} variant='filled'>
-                            <InputLabel htmlFor='country' color='secondary' className='!text-text'>
+                        <FormControl sx={{ m: 0.5, width: 210 }} variant='filled'>
+                            <InputLabel
+                                htmlFor='country-input'
+                                color='secondary'
+                                className='!text-text'
+                            >
                                 Change Country
                             </InputLabel>
-                            <FilledInput
+                            <Select
+                                id='country-input'
                                 name='country'
+                                color='secondary'
+                                className='!text-text text-left'
                                 value={country}
-                                onChange={(e) => {
-                                    setCountry(e.target.value);
-                                }}
                                 error={countryError}
+                                onChange={(e) => setCountry(e.target.value)}
                                 onFocus={() => setCountryError(false)}
-                                inputProps={{ maxLength: 2, minLength: 2 }}
-                                sx={{ m: 0.5, width: 210 }}
-                            />
+                                MenuProps={{ PaperProps: { sx: { maxHeight: 300 } } }}
+                                defaultValue={country}
+                            >
+                                {COUNTRIES.map((item, i) => (
```

Comment:
I think we could memoize this and avoid calling map in the render method to improve performance.

Reply:
@Thenlie Memoizing this creates a weird behavior with the `<Select />` component and the `onChange` prop. Although event target values are logged appropriately, modifying states doesn't trigger the `value` to update. I believe it's an async issue. Any insight on this?

Reply:
I am not entirely sure how you had implemented this, but I think I would do something like this:

```diff
diff --git a/src/components/modals/EditProfileModal.tsx b/src/components/modals/EditProfileModal.tsx
index e04a0a7..967e5bc 100644
--- a/src/components/modals/EditProfileModal.tsx
+++ b/src/components/modals/EditProfileModal.tsx
@@ -1,4 +1,4 @@
-import React, { useState } from 'react';
+import React, { useMemo, useState } from 'react';
 import {
     updateProfileUsername,
     setProfileAdultFlag,
@@ -86,6 +86,14 @@ const EditProfileModal: React.FC<EditProfileModalProps> = ({
         }
     };
 
+    const DropDownItems: JSX.Element[] = useMemo(() => {
+        return COUNTRIES.map((item, i) => (
+            <MenuItem key={i} value={item.country}>
+                {item.country}
+            </MenuItem>
+        ));
+    }, [COUNTRIES]);
+
     return (
         <>
             <Button title='Edit Profile' startIcon={<Edit />} onClick={handleOpen} />
@@ -154,11 +162,7 @@ const EditProfileModal: React.FC<EditProfileModalProps> = ({
                                 MenuProps={{ PaperProps: { sx: { maxHeight: 300 } } }}
                                 defaultValue={country}
                             >
-                                {COUNTRIES.map((item, i) => (
-                                    <MenuItem key={i} value={item.country}>
-                                        {item.country}
-                                    </MenuItem>
-                                ))}
+                                {DropDownItems}
                             </Select>
                         </FormControl>
                         <Typography>Current Country: {profile?.country}</Typography>
```